File: 1-two-sum/1-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        d={}
        k=target
        n=len(nums)
        res=[0,0]
        for i in range(0,n):
            if nums[i] not in d:
                d[k-nums[i]]=i
            else:
                res[0]=d[nums[i]]
                res[1]=i
                break
        #print(res)
        return res
        

File: 1-two-sum/1-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int x) {
        int n=nums.size();
        map<int,int> mp;
        vector<int> res;
        for(int i=0;i<n;i++)
        {
            if(mp.find(x-nums[i])!=mp.end())
            {
                res.push_back(mp[x-nums[i]]);
                res.push_back(i);
                break;
            }
            mp[nums[i]]=i;
        }
        return res;
    }
};